src/transformers/modeling_attn_mask_utils.py

```diff
@@ -227,12 +229,24 @@ def _unmask_unattended(
         ```
         """
         # fmt: on
-        if expanded_mask.dtype == torch.bool:
-            raise ValueError(
-                "AttentionMaskConverter._unmask_unattended expects a float `expanded_mask`, got a BoolTensor."
-            )
 
-        return expanded_mask.mul(~torch.all(expanded_mask == min_dtype, dim=-1, keepdim=True))
+        if attention_mask is not None and attention_mask.device.type == "cuda":
+            if expanded_mask.dtype == torch.bool:
+                raise ValueError(
+                    "AttentionMaskConverter._unmask_unattended expects a float `expanded_mask`, got a BoolTensor."
+                )
+
+            # TODO: For dynamo, rather use a check on fullgraph=True once this is possible (https://github.com/pytorch/pytorch/pull/120400).
+            is_tracing = torch.jit.is_tracing() or (hasattr(torch, "_dynamo") and torch._dynamo.is_compiling())
+            is_tracing |= isinstance(input_tensor, torch.fx.Proxy)
```

Comment:
You can leave this as a one-liner. I believe in python `a or b` never evaluates b if a is already True.